### PR TITLE
add bad datatype checking to POST projects/

### PR DESCRIPTION
**Pull request by tschuy**
Currently if you POST a string to the slug field, it'll crash the server when it attempts to run a ``.filter`` on it. Check your datatypes, folks!

### CODE REVIEW

**Comment by tschuy** [comment]
@Kennric

**Comment by tschuy** [comment]
Fixes #137.

**Comment by Kennric** [comment]
This seems to address the issue, and I don't see any problems. It should be noted that the tests check by sending a single wrong type to the endpoint, so it isn't a comprehensive test for all possible bad types, but verifies that a single wrong type is caught. I don't think that's a problem at this point, based on how the code is doing the check. +1

File: src/projects.js [hunk 1]
```diff
@@ -171,6 +177,30 @@ module.exports = function(app) {
...
+      if (validationFailure) {
+        const err = errors.errorBadObjectInvalidField('project',
+          validationFailure.name, validationFailure.type,
+          validationFailure.actualType);
+        return res.status(err.status).send(err);
+      }
+
+      // check validity of uri syntax
+      if (obj.uri && !validUrl.isWebUri(obj.uri)) {
+        const err = errors.errorInvalidIdentifier('uri', obj.uri);
```
Comment (LyonesGamer) [inline src/projects.js]: This needs to be changed to a Bad Object error (type Invalid Field).

**Comment by LyonesGamer** [comment]
+1

**Comment by Kennric** [comment]
Good catch on the object errors @LyonesGamer  +1

**Pull request merged**